pandas/core/arrays/_ranges.py

```diff
@@ -150,7 +156,12 @@ def _generate_range_overflow_safe_signed(
         addend = np.int64(periods) * np.int64(stride)
         try:
             # easy case with no overflows
-            return np.int64(endpoint) + addend
+            result = np.int64(endpoint) + addend
+            if result == iNaT:
```

Comment:
can't this wrap around? (i mean you don't want it too)

Reply:
the np.errstate on L155 prevents that